documentation/src/main/asciidoc/topics/ref_discovery_tcp_ping.adoc

```diff
@@ -9,7 +9,11 @@ With TCPPING, you manually specify the IP address or hostname of each node in th
 ----
 include::config_examples/tcpping.xml[]
 ----
-<1> For reliable discovery, Red Hat recommends `port-range=0`.
+
+[NOTE]
+====
+`port-range=0` is recommended for reliable discovery.
```

Comment:
This is not ok. If `TCP.port-range="10"` (JGroups default), `TCPPING.port-range="0"`, and port `bind_port` is busy, then TCP will bind to port `bind_port+1`, but TCPPING on the other nodes will ignore it, making discovery *less* reliable.

We could say instead that setting `<TCP port-range="0"/>` is recommended in production, unless multiple nodes are expected to use the same bind address and bind port. But this recommendation would be just as valid with TCP+MPING or TCP+DNS_PING, not just for TCP+TCPPING.

Reply:
@danberindei Thanks for catching this one. Would it be sense to remove this note altogether? 

This seems like something that might be a tuning recommendation for TCP stacks. I'm also trying to understand this a bit better. 

```
<TCP bind_port="7800"  port_range="0"/>
<TCPPING timeout="3000" initial_hosts="${jgroups.tcpping.initial_hosts:HostA[7800],HostB[7801]}" port_range="1"/>
```

In this example, if `bind_port` is busy then TCP cannot bind to it. In this case, I think an exception is thrown and the node doesn't start. 

IIRC this recommendation came from @wfink. I think setting `port_range="0"` for TCPPING means discovery happens with the initial_hosts only.

Reply:
The issue here is that having a port_range no matter whether it is TCP or TCP_PING jgroups dont fail if the port is used for TCP and use the next free port.
This can lead in confusing behavior where a node is no found with PING or PING will find nodes which are not meant to join the cluster.
Because of this we recommend for EAP and should do the same here.

Note that there might be an EAP and DG cluster on the same machine and JGroups will detect them

Reply:
@wfink Thanks. So `port_range="0"` basically restricts JGroups to either the `bind_port` (TCP) or `initial_hosts` (TCPPING)? 

This is necessary to ensure that only nodes that are meant to join the cluster can join the cluster, keeping EAP nodes separate from DG clusters, etc.

If I understand correctly this info would probably be better as a standalone topic, instead of as a note in the TCPPING docs. Please let me know and I'll create a separate Jira for that.

Reply:
Yes, it will effect the 'normal local' cluster detection and as well the xsite separation.
The port_range might cause issues if both stacks will use TCP/TCP_PING so the automatic increment can overlap.
As well as two distinct clusters which should not see each other like dev/test or worse dev/test/prod
So separate Jira seems fine

Reply:
@wfink Cheers. I've created https://issues.redhat.com/browse/ISPN-12786